### ISSUE TITLE
Using the network port to upload the program requires three questions such as entering a password

### ISSUE THREAD

**Issue by Loryoung** [arduino/arduino-ide#1770]
## Problem 1

UPDATE: **I found that the problem has been fixed in the nightly build version**

<s>

### Describe the problem

IDE remains in uploading state after successful upload to network port

### To reproduce

1. Upload to the network port of an ESP8266 board.

🐛 It will always prompt that the upload is in progress, but the program has actually been uploaded successfully, and the ESP8266 program has been successfully updated.

![image](https://user-images.githubusercontent.com/53025669/207584531-8be7e350-938c-4b91-96ef-04a6ac707b78.png)

Failure to prompt that the upload is complete will result in the inability to click the upload button again.

![image](https://user-images.githubusercontent.com/53025669/207586523-34170945-9f07-4126-a5d5-c8df3f1ca727.png)

### Expected behavior

Once the program upload is complete it should complete normally.

### Additional context

Upload operation completes as expected when using Arduino IDE 1.8.19

</s>

### Describe the problem

If no password is configured, the password is still required when uploading to a network port.

### To reproduce

1. Set up an ESP8266 board for OTA uploads without password verification

🐛 Arduino IDE still prompts for a password at the start of the upload.

![image](https://user-images.githubusercontent.com/53025669/207586073-dc7b2ec9-b209-49a9-bbdf-d3001f0971e6.png)

### Expected behavior

If no password is configured, no password is required.

### Additional context

Arduino IDE 1.x does not ask for a password if no password is configured.

### Describe the problem

The new version of the IDE does not display some examples

The ArduinoOTA-related examples for ESP8266 are not displayed.

![image](https://user-images.githubusercontent.com/53025669/207589816-89127ea8-d1b3-472f-a0fb-7dff7574d9f4.png)

### Expected behavior

The IDE should show all routines.

### Additional context

The examples are shown in Arduino IDE 1.x:

![image](https://user-images.githubusercontent.com/53025669/207589917-8e8e2b73-bdc9-40a6-9eb0-e0f5bb8d4725.png)

---

### Arduino IDE version

3.0.3

### Operating system

Windows

### Operating system version

Windows 10

### Additional context

testing platform：NodeMCU 1.0 (ESP-12E Module)

![image](https://user-images.githubusercontent.com/53025669/207585016-58a9fdf0-e9cf-4002-ab07-f7a25e91d2af.png)

test program：BasicOTA.ino

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-ide/issues?q=)
- [X] I verified the problem still occurs when using the latest [nightly build](https://www.arduino.cc/en/software#nightly-builds)
- [X] My report contains all necessary details

**Comment by per1234** [comment]
Hi @Loryoung. Thanks for taking the time to submit an issue.

>IDE remains in uploading state after successful upload to network port

Even though you said this was fixed for you in the nightly build, the different result may have instead been caused by the intermittent nature of the bug because no specific fix has been made since the 2.0.3 release (though it is possible it was fixed as a side effect of another change). We already have a dedicated issue tracking this here: https://github.com/arduino/arduino-ide/issues/1356

>If no password is configured, the password is still required when uploading to a network port.

Unfortunately I don't think there is any way for the behavior of Arduino IDE 1.x to be restored. The problem is Arduino IDE 1.x's network port upload system was very application-specific. This has been replaced by a flexible generalized system that allows uploading to any arbitrary protocol of port (instead of only supporting serial and network ports as was the case in Arduino IDE 1.x). This new system means that Arduino IDE 2.x doesn't have a way of getting the data from the board about whether the user provided fields are required by the firmware on the board or not. It only knows about the fields that are specified for the port's protocol by the platform.

However, I do think the user experience in this case can be improved. My proposal for doing so is tracked here: https://github.com/arduino/arduino-ide/issues/1488

> The ArduinoOTA-related examples for ESP8266 are not displayed.

I see we have another report about this at https://github.com/arduino/arduino-cli/issues/1656.

It is best to have only a single issue per subject so we can consolidate all relevant discussion to one place, so I'll go ahead and close this in favor of the other.

If you end up with additional information to share, feel free to comment in the other thread.

---

@Loryoung I have a request for you: in the future, please submit a separate issue for each distinct defect or feature request you wish to report. These issues that contain multiple subjects are very difficult for us to maintain and make it more likely that we will forget to resolve one of the items.
